src/Oscoin/Data/Ledger.hs

```diff
@@ -225,6 +259,28 @@ deriving instance (Eq (Crypto.Hash c), Eq (AccountId c))     => Eq   (Dependency
 deriving instance (Ord (Crypto.Hash c), Ord (AccountId c))   => Ord  (DependencyUpdate c)
 deriving instance (Show (Crypto.Hash c), Show (AccountId c)) => Show (DependencyUpdate c)
 
+instance ( Serialise (Crypto.Hash c)
+         , Serialise (AccountId c)
+         , Serialise (Signature c)
+         ) => Serialise (DependencyUpdate c)
+  where
+    encode (Depend proj hsh) =
+           CBOR.encodeListLen 3
+        <> CBOR.encodeWord 0
+        <> CBOR.encode proj
+        <> CBOR.encode hsh
+    encode (Undepend proj) =
+           CBOR.encodeListLen 2
+        <> CBOR.encodeWord 1
+        <> CBOR.encode proj
+
+    decode = do
+        pre <- liftA2 (,) CBOR.decodeListLen CBOR.decodeWord
```

Comment:
Apparently also the list len has its own canonical version -- `decodeListLenCanonical`